Source/engine/palette.cpp

```diff
@@ -360,44 +360,41 @@ void palette_update_caves()
 	palette_update(0, 31);
 }
 
+/**
+ * @brief Cycle the lava every other frame, and glow every frame
+ * Lava has 15 colors and the glow 16, so the full animation has 240 frames before it loops
+ */
 void palette_update_crypt()
 {
-	static int laveDelay = 0;
-	static int glowDelay = 0;
+	static bool delayLava = false;
 
-	if (laveDelay > 1) {
+	if (!delayLava) {
 		CycleColorsReverse(1, 15);
-		laveDelay = 0;
-	} else {
-		laveDelay++;
-	}
-	if (glowDelay > 0) {

```

Comment:
I think this doesn't matter but to be safe:
Removing this is not 100% the same.
In this pr glow will always cycle. In this master this is not always the case
- In the first run `glowDelay` is 0 and we don't cycle.
- `glowDelay` is never rest and that means after some time int overflows and we get back to a `glowDelay` of 0 that don't cycle.

Reply:
Considering that the stutter only happens every 500 day of being in crypts with out restarting Diablo I consider that more of a bug then anything :)